src/main/java/com/google/api/codegen/discovery/transformer/py/PythonSampleTypeNameConverter.java

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.transformer.py;
+
+import com.google.api.codegen.discovery.config.TypeInfo;
+import com.google.api.codegen.discovery.transformer.SampleTypeNameConverter;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypedValue;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Field;
+
+/** Maps SampleConfig and TypeInfo instances to Python specific TypeName instances. */
+class PythonSampleTypeNameConverter implements SampleTypeNameConverter {
+
+  /** A map from primitive types in proto to Python counterparts. */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_TYPE_MAP =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_BOOL, "dict")
+          .put(Field.Kind.TYPE_INT32, "int")
+          .put(Field.Kind.TYPE_INT64, "long")
+          .put(Field.Kind.TYPE_UINT32, "int")
+          .put(Field.Kind.TYPE_UINT64, "long")
+          .put(Field.Kind.TYPE_FLOAT, "float")
+          .put(Field.Kind.TYPE_DOUBLE, "float")
+          .put(Field.Kind.TYPE_STRING, "str")
+          .put(Field.Kind.TYPE_ENUM, "str")
+          .build();
+
+  /** A map from primitive types in proto to zero value in Python. */
+  private static final ImmutableMap<Field.Kind, String> PRIMITIVE_ZERO_VALUE =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_UNKNOWN, "{}")
+          .put(Field.Kind.TYPE_BOOL, "False")
+          .put(Field.Kind.TYPE_INT32, "0")
+          .put(Field.Kind.TYPE_INT64, "str(0L)")
+          .put(Field.Kind.TYPE_UINT32, "0")
+          .put(Field.Kind.TYPE_UINT64, "str(0L)")
+          .put(Field.Kind.TYPE_FLOAT, "0.0")
+          .put(Field.Kind.TYPE_DOUBLE, "0.0")
+          .put(Field.Kind.TYPE_STRING, "''")
+          .put(Field.Kind.TYPE_ENUM, "''")
+          .build();
+
+  public PythonSampleTypeNameConverter() {}
+
+  @Override
+  public TypeName getServiceTypeName(String apiTypeName) {
```

Comment:
Since we're implementing an interface it can't be avoided... it could be converted to an abstract class but there isn't necessarily much common ground for this functions between languages.

As part of this 2nd refactor round I'm working on I intend to reduce the complexity of this class as the `TypeNameGenerator` goes away, let me come back to this.

#795